frontend/src/containers/PostCreationView.tsx

```diff
@@ -0,0 +1,87 @@
+import PostForm from "components/PostForm";
+import React, {useEffect, useState} from "react";
+import {PostData, submitPost} from "actions/posts";
+import {Web3Context} from "reducers/bc";
+import {State} from "reducers/index";
+import {connect} from "react-redux";
+import {initWeb3} from "actions/bc";
+import {redirectIfNotAuthenticated} from "src/utils";
+import {RouteComponentProps} from 'react-router-dom';
+
+interface StateToProps {
+    web3Context: Web3Context | null,
+    submitLoading: boolean,
+    submitError: Error | null,
+}
+
+interface DispatchToProps {
+    initWeb3: typeof initWeb3,
+    submitPost: typeof submitPost,
+}
+
+type Props = StateToProps & DispatchToProps & RouteComponentProps;
+
+const PostCreationComponent = (props: Props) => {
+    const [wasLoading, setWasLoading] = useState(false);
+
+    let error = null;
+    let loading = null;
+    useEffect(() => {
+        if (props.submitError) {
+            error = (<div>Omg, something went wrong!</div>);
```

Comment:
I think more natural approach would be to have a error component, and some flag that indicates if error happenend, and rendering error in that case